### PR TITLE
[API] Training API with generator @open sesame 05/29 20:16

### PR DESCRIPTION
**Pull request by jijoongmoon**
There is no API to get training, validation, test data from generator
function.

In this PR, introduces new training API with generator function.
. New Tizen API.
. Add Unit Test API with generator function.
. Add Test generator for future test.

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>

### CODE REVIEW

File: test/nntrainer_test_util.cpp [hunk 1]
```diff
@@ -48,3 +58,189 @@ void replaceString(const std::string &from, const std::string &to,
...
+ * @param[out] outLabel
+ * @param[in] id th data to get
+ * @retval true/false false : end of data
+ */
+static bool getData(std::ifstream &F, std::vector<float> &outVec,
+                    std::vector<float> &outLabel, int id) {
+  F.clear();
+  F.seekg(0, std::ios_base::end);
+  uint64_t file_length = F.tellg();
+  uint64_t position = (feature_size + num_class) * id * sizeof(float);
```
Comment (zhoonit) [inline test/nntrainer_test_util.cpp:91]: I think, it would be nice to have handle the case when `id < 0`

File: test/nntrainer_test_util.cpp [hunk 2]
```diff
@@ -22,8 +22,18 @@
...
 #include <iostream>
 
+#define num_class 10
+#define mini_batch 16
+#define feature_size 62720
+
+bool *duplicate;
+bool *valduplicate;
+bool alloc_train = false;
+bool alloc_val = false;
```
Comment (dongju-chae) [inline test/nntrainer_test_util.cpp:35]: How about appending `static`?

File: test/nntrainer_test_util.cpp (hunk 1 above)
Comment (jijoongmoon) [reply test/nntrainer_test_util.cpp:93]: fixed. Thanks.

File: test/nntrainer_test_util.cpp (hunk 2 above)
Comment (jijoongmoon) [reply test/nntrainer_test_util.cpp]: fixed. Thanks.

**Pull request merged**